doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+{
+    ...
+
+    /** Insertion of Segment Route Policy with Reduced SRH */
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_INSERT_RED,
+
+    ...
+
+    /** Encapsulation in a Segment Route Policy with Reduced SRH */
+    SAI_SEGMENTROUTE_SIDLIST_TYPE_ENCAPS_RED,
+
+
+} sai_segmentroute_sidlist_type_t;
+``` 
+
+2. Added enum sai_local_sid_entry_endpoint_type_t to provide the initial list of endpoint behaviors.
+
+```
+/**
+ * @brief Enum defining Endpoint Action types
+ */
+typedef enum _sai_local_sid_entry_endpoint_type_t
+{
+    /** Basic Endpoint */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_E,
+
+    /** End.X Endpoint with Layer-3 Cross-connect */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X,
+
+    /** End.T Endpoint with specific IPv6 Table */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T,
+
+    /** Endpoint with decapsulation and IPv6 Cross-connect */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DX6,
+
+    /** Endpoint with decapsulation and IPv4 Cross-connect */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DX4,
+
+    /** Endpoint with decapsulation and specific IPv6 table lookup */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6,
+
+    /** Endpoint with decapsulation and specific IPv6 table lookup */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4,
+
+    /** Endpoint with decapsulation and specific IP table lookup */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46,
+
+    /** Endpoint Bound to a policy with Encapsulation */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_ENCAPS,
+
+    /** End.B6.Encaps function with a reduced SRH */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_ENCAPS_RED,
+
+    /** Endpoint Bound to a policy with Insertion */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_INSERT,
+
+    /** End.B6.Insert function with a reduced SRH */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_INSERT_RED,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of Custom range base */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_endpoint_type_t;
+```
+3. Added enum sai_local_sid_entry_endpoint_flavor_t to list all the combinations of endpoint flavors 
+   as defined in RFC-8986
+
+```
+/**
+ * @brief Enum defining Endpoint Segment flavors for End, End.X and End.T functions
+ */
+typedef enum _sai_local_sid_entry_endpoint_flavor_t
+{
+    /** Penultimate segment pop */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP,
+
+    /** Ultimate Segment pop */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USP,
+
+    /** Ultimate Segment decapsulation */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USD,
+
+    /** PSP and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP,
+
+    /** USD and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USD_AND_USP,
+
+    /** PSP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USD,
+
+    /** PSP, USP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP_AND_USD
+
+    /** None */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_NONE
+
```

Comment:
For uSID, will need "Shift-only" flavour as well.

Reply:
For uSID, I was thinking about add uN and uA behaviors in a future PR. For uN/uA, the shift-and-lookup/shift-and-forward functionality can be implicit. Can you please explain where shift-only flavor will be required?

Reply:
I've added uN and uA behaviors for uSID. My understanding is that shift-only behavior needs to be advertised in the control plane by the legacy devices but I haven't found a need for it at the SAI level.